impl/src/planning.rs

```diff
@@ -416,82 +469,122 @@ impl<'planner> WorkspaceSubplanner<'planner> {
+    let mut catalogs = vec![self.crate_catalog];
+    for catalog in self.binary_dependencies.iter() {
+      catalogs.push(catalog);
+    }
+
+    // Build a list of all workspace files
+    let mut files: Vec<&CargoWorkspaceFiles> = self
+      .binary_deps_files
+      .iter()
+      .map(|(_key, val)| val)
+      .collect();
+    files.push(self.files);
+
     // Gather the checksums for all packages in the lockfile
     // which have them.
     //
     // Store the representation of the package as a tuple
     // of (name, version) -> checksum.
     let mut package_to_checksum = HashMap::new();
-    if let Some(lock_path) = self.files.lock_path_opt.as_ref() {
-      let lockfile = Lockfile::load(lock_path.as_path())?;
-      for package in lockfile.packages {
-        if let Some(checksum) = package.checksum {
-          package_to_checksum.insert(
-            (package.name.to_string(), package.version),
-            checksum.to_string(),
-          );
+    for workspace_files in files.iter() {
+      if let Some(lock_path) = workspace_files.lock_path_opt.as_ref() {
+        let lockfile = Lockfile::load(lock_path.as_path())?;
+        for package in lockfile.packages {
+          if let Some(checksum) = package.checksum {
+            package_to_checksum.insert(
+              (package.name.to_string(), package.version),
+              checksum.to_string(),
+            );
+          }
         }
       }
     }
 
-    self
-      .metadata
-      .resolve
-      .as_ref()
-      .ok_or_else(|| RazeError::Generic("Missing resolve graph".into()))?
-      .nodes
+    // Additionally, the binary dependencies need to have their checksums added as well
+    for (pkg, info) in self.settings.binary_deps.iter() {
+      let version = cargo_lock::Version::parse(info.req())?;
+      package_to_checksum.insert(
+        (pkg.clone(), version.clone()),
+        fetch_crate_checksum(&self.settings.index_url, pkg, &version.to_string())?,
+      );
+    }
+
+    let contexts = catalogs
       .iter()
-      .sorted_by_key(|n| &n.id)
-      .filter_map(|node| {
-        // UNWRAP: Node packages guaranteed to exist by guard in `produce_planned_build`
-        let own_crate_catalog_entry = self.crate_catalog.entry_for_package_id(&node.id).unwrap();
-        let own_package = own_crate_catalog_entry.package();
-
-        let checksum_opt =
-          package_to_checksum.get(&(own_package.name.clone(), own_package.version.clone()));
-
-        // Skip the root package (which is probably a junk package, by convention)
-        if own_crate_catalog_entry.is_root() {
-          return None;
-        }
+      .map(|catalog| {
+        (*catalog)
+          .metadata
+          .resolve
+          .as_ref()
+          .ok_or_else(|| RazeError::Generic("Missing resolve graph".into()))?
+          .nodes
+          .iter()
+          .sorted_by_key(|n| &n.id)
+          .filter_map(|node| -> Option<Result<CrateContext>> {
```

Comment:
Move this filter_map body somewhere else or refactor it to be shorter.

Reply:
https://github.com/google/cargo-raze/pull/227#discussion_r503446722 
sorry 😅 fixed


https://github.com/google/cargo-raze/pull/227#discussion_r503446934
I added `create_crate_context`